### PR TITLE
Clean up register preservation requirements for Enter callback

### PR DESCRIPTION
**Pull request by noahfalk**
#19023 

I'm still testing this, so adding NO_MERGE tag even if CI suggests it is good.

@BruceForstall - PTAL
cc @sergign60

### CODE REVIEW

File: src/vm/amd64/asmhelpers.S [hunk 1]
```diff
@@ -52,16 +52,9 @@
...
-//           - xmm0, xmm1
-//
-//       Not Preserved:
-//           - integer argument registers (rcx, rdx, r8, r9)
-//           - floating point argument registers (xmm1-3)
-//           - volatile integer registers (r10, r11)
-//           - volatile floating point registers (xmm4-5)
-//           - upper halves of ymm registers on AVX (which are volatile)
+//           - All argument and callee-saved registers must be preserved:
+//             RDI, RSI, RDX, RCX, R8, R9,  RBX, RBP, R12�R15, XMM0, XMM1, XMM2, XMM3, XMM4, XMM5, XMM6, XMM7
```
Comment (BruceForstall) [inline src/vm/amd64/asmhelpers.S]: The character between R12 and R15 shows up as garbage when I view this in CodeFlow; I wonder if it's some Unicode thing.

**Review by BruceForstall — APPROVED** [review]
LGTM. Thanks for the detailed addition to CLR-ABI.

**Comment by BruceForstall** [comment]
@noahfalk I presume you're still working on this...

**Comment by noahfalk** [comment]
I set it down for a while. Eventually I'll get back to it unless someone wants to beat me to it. The remaining work is testing.

**Comment by BruceForstall** [comment]
@noahfalk Just checking... you still expect to get back to it?

**Comment by noahfalk** [comment]
Good call … I had completely forgotten about this : ) I'd still like to get this in but I should probably be more realistic, it may not be me that does it and it may not be soon. Any particular reason you were asking or just checking on status?

**Comment by BruceForstall** [comment]
Just checking. I've been looking through lots of old issues/PRs seeing if there's old stuff that can be closed.

**Comment by BruceForstall** [comment]
@echesakovMSFT @noahfalk @davmason Is this PR still valid? Presumably at least the CLR-ABI.md changes could be checked in (if they are still correct)?

**Comment by echesakovMSFT** [comment]
@echesakovMSFT I can work on this

**Comment by maryamariyan** [comment]
Thank you for your contribution. As announced in dotnet/coreclr#27549 this repository will be moving to dotnet/runtime on November 13. If you would like to continue working on this PR after this date, the easiest way to move the change to dotnet/runtime is:
1.	In your coreclr repository clone, create patch by running `git format-patch origin`
2.	In your runtime repository clone, apply the patch by running `git apply --directory src/coreclr <path to the patch created in step 1>`

**Comment by maryamariyan** [comment]
Thank you for your contribution. As announced in #27549 the dotnet/runtime repository will be used going forward for changes to this code base. Closing this PR as no more changes will be accepted into master for this repository. If you’d like to continue working on this change please move it to dotnet/runtime.

**Pull request closed without merging**